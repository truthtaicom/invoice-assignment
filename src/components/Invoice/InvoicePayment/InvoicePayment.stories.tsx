import { storiesOf } from '@storybook/react';
import React from 'react';
import InvoicePayment from "./InvoicePayment"
import { simpleInvoiceListData } from '../InvoiceList/InvoiceList.utils'

storiesOf("Component", module)
  .add("Invoice Payment", () => (
    <InvoicePayment data={simpleInvoiceListData} selected={{ id: 1 }} />
  ))