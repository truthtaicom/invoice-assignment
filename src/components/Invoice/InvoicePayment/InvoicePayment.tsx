import React, { useState } from 'react'
import styled from 'styled-components';
import InvoiceItem, { IInvoiceItem } from '../InvoiceList/InvoiceItem'
import { IBANInput, SimpleList, Form, FormItem, Button, Text } from '../../../ui-kit'

export interface IInvoicePayment{
  onSelect?: (e: any) => void,
  onSubmit?: (e: any) => void,
  onChangeIBANNum?: (e: any) => void,
  data: IInvoiceItem[],
  selected?: IInvoiceItem
}


const StyledInvoiceInformation = styled.div`
  display: grid;
  padding: 1rem;
`

const StyledInvoicePaymentForm = styled(Form)`
  display: grid;
  grid-row-gap: 1rem;
`

const StyledFoundInvoiceText = styled(Text)``

const StyledBtnConfirm = styled(Button)`
  margin: 0;
`

const InvoicePayment: React.FC<IInvoicePayment> = React.memo((props) => {
  const onChangeIBAN = (e) => {
    const value = e.target.value
    if (!value.includes('_')) {
      props.onChangeIBANNum && props.onChangeIBANNum(value)
    }
  }

  const onSelectItem = (values) => {
    props.onSelect && props.onSelect(values)
  }

  return (
    <StyledInvoiceInformation>
      <StyledInvoicePaymentForm onSubmit={props.onSubmit}>
        <FormItem>
          <IBANInput
            name="ibanNum"
            label="Search IBAN"
            onChange={onChangeIBAN}
          />
        </FormItem>
        {
          props.data && props.data.length > 0
          && (
            <>
              <StyledFoundInvoiceText>Found {props.data.length} payment(s) from bank transfer</StyledFoundInvoiceText>
              <SimpleList data={props.data} renderItem={
                (itemProps) => <InvoiceItem actived={itemProps.id === (props.selected && props.selected.id)} {...itemProps} onClick={() => onSelectItem(itemProps)} />} 
              />
            </>
          )
        }

        {
          props.data && props.data.length <= 0 && (<StyledFoundInvoiceText>Not found any payments</StyledFoundInvoiceText>)
        }
        
        
        <StyledBtnConfirm color="primary" disabled={!props.selected} type="submit">Done</StyledBtnConfirm>
      </StyledInvoicePaymentForm>
    </StyledInvoiceInformation>
  )
})

export default InvoicePayment
