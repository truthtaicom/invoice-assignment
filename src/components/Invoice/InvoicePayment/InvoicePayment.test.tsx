import React from 'react'
import { render } from 'react-testing-library'
import InvoicePayment from './InvoicePayment'

describe('<InvoicePayment /> spec', () => {
  it('renders the component', () => {
    const component = render(<InvoicePayment data={[{ amount: 1 }]}/>)
    expect(component).toMatchSnapshot()
  })
});