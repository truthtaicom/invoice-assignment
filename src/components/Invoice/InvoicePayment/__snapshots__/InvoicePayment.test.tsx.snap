// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvoicePayment /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-brqgnP bGWdk"
      >
        <form
          class="sc-cMljjf jWaZMD sc-eNQAEJ fYukrM"
        >
          <div
            class="sc-hMqMXs dSDiJQ"
          >
            <div
              class="sc-htpNat cWnILl"
            >
              <div
                class="sc-bxivhb iiUfTC"
              >
                <span
                  class="sc-gzVnrw gtppSg"
                >
                  Search IBAN
                </span>
                <span
                  class="sc-gzVnrw gtppSg"
                >
                   
                </span>
                <span
                  class="sc-gzVnrw gtppSg"
                >
                  +
                </span>
              </div>
              <div
                class="sc-bxivhb evsUlz"
              >
                <label
                  class="sc-ifAKCX hSWqAs"
                >
                   
                  <span
                    class="sc-EHOje cZmnwG"
                  >
                    Enter 
                  </span>
                    
                  Search IBAN
                </label>
                <input
                  class="sc-bZQynM htDgzj"
                  name="ibanNum"
                  placeholder="DE-12-1234-1234-12345-12"
                />
              </div>
            </div>
          </div>
          <button
            class="sc-bdVaJa sc-jDwBTQ iKfuNx"
            color="primary"
            disabled=""
            type="submit"
          >
            Done
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-brqgnP bGWdk"
    >
      <form
        class="sc-cMljjf jWaZMD sc-eNQAEJ fYukrM"
      >
        <div
          class="sc-hMqMXs dSDiJQ"
        >
          <div
            class="sc-htpNat cWnILl"
          >
            <div
              class="sc-bxivhb iiUfTC"
            >
              <span
                class="sc-gzVnrw gtppSg"
              >
                Search IBAN
              </span>
              <span
                class="sc-gzVnrw gtppSg"
              >
                 
              </span>
              <span
                class="sc-gzVnrw gtppSg"
              >
                +
              </span>
            </div>
            <div
              class="sc-bxivhb evsUlz"
            >
              <label
                class="sc-ifAKCX hSWqAs"
              >
                 
                <span
                  class="sc-EHOje cZmnwG"
                >
                  Enter 
                </span>
                  
                Search IBAN
              </label>
              <input
                class="sc-bZQynM htDgzj"
                name="ibanNum"
                placeholder="DE-12-1234-1234-12345-12"
              />
            </div>
          </div>
        </div>
        <button
          class="sc-bdVaJa sc-jDwBTQ iKfuNx"
          color="primary"
          disabled=""
          type="submit"
        >
          Done
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
